src/Shared/interfaces.ts

```diff
@@ -200,6 +200,7 @@ export interface PageEditProps {
 export interface DashboardProps {
   index: IndexReadyItem[];
   isComplete: boolean;
```

Comment:
It is a pity that we now have to pass in two props just to switch one button. How about using a three-way enumeration:

```
enum AddStatus {
  Add,
  Wait,
  Full,
}
```

and then pass in just one prop `addStatus: AddStatus` instead of `isComplete` and `newTranslationStatus` with these benefits:

- Safer contract. The calling Controller knows exactly what it can and can't pass in.
- We cut out the logic from the template. We can exercise the business rule on the Adonis side.